src/ProxyClient/Nginx.php

```diff
@@ -49,7 +49,7 @@ public function setPurgeLocation($purgeLocation = '')
     public function refresh($url, array $headers = [])
     {
         $headers = array_merge($headers, [self::HTTP_HEADER_REFRESH => '1']);
-        $this->queueRequest(self::HTTP_METHOD_REFRESH, $url, $headers);
+        $this->httpAdapter->invalidate($this->messageFactory->createRequest(self::HTTP_METHOD_REFRESH, $url, $headers));
```

Comment:
looking at this, we could also keep the queueRequest to make this less verbose. wdyt @ddeboer ?